### PR TITLE
fixed multicast plugin bug and added test

### PR DESCRIPTION
**Pull request by bilalyasar**
I have rebased and opened new pr from this pr:
https://github.com/hazelcast/hazelcast/pull/8421

- added javadoc
- fixed small bug ( discovery property bug, invalid port check)
- added test

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/spi/discovery/multicast/MemberToMemberDiscoveryTest.java [hunk 1]
```diff
@@ -47,7 +48,9 @@ public void setUp() {
 
         System.setProperty(TestEnvironment.HAZELCAST_TEST_USE_NETWORK, "true");
         System.setProperty("java.net.preferIPv4Stack", "true");
-        factory = createHazelcastInstanceFactory(2);
+        if (factory == null) {
```
Comment (mesutcelik) [inline hazelcast/src/test/java/com/hazelcast/spi/discovery/multicast/MemberToMemberDiscoveryTest.java]: Do we really need this? `factory` is already defined in HazelcastTestSupport so all cleanup, null checking etc. looks unnecessary to me. 

I believe we should remove `factory` variable in this class and depend on the one @ HazelcastTestSupport. Good cleanup opportunity for you 👍

**Comment by mesutcelik** [comment]
👍 with minor comment.

Can you please add bug description as well? invalid port fix etc.

**Comment by bilalyasar** [comment]
@mesutcelik reviewed your comments, also added description.

**Comment by Donnerbart** [comment]
:+1:

**Pull request merged**